core/relay/src/services/network_service.rs

```diff
@@ -48,6 +48,7 @@ impl Network for NetworkService {
                          );
                 match request_state.state.as_ref() {
                     Some(state) => {
+                        try_mark_request_state_deleted(request_state.clone(), request_id.to_string(), db);
```

Comment:
Add comment here that we delete only if the view was received and client also has got the view.